### DISCUSSION TITLE
Without installation will it work?

### DISCUSSION THREAD

**Discussion by semibreave** [dfinke/ImportExcel#1589] [Q&A]
I wrote a script that will fetch data from server and put it in excel spreadsheet.we would like to have the user to run the script themselves on their servers.The challenge is whether should everyone install the module on their server before running or we should set up a centralized machine with importexcel installed and get everyone to run from there?

**Comment by dfinke** [comment]
Whereever functions are run, that environment needs access to the PowerShell Excel module and your script. If you can centralize it, that is best. That way when you update your script, everyone gets the update. Same goes for the Excel module. If it is centralized and it gets updated, everyone gets the new version. Saves on having to intall it everywhere.

The downside is, if you change your script and it has errors in it etc. You have to quickly fix or roll back.

**Comment by semibreave** [comment]
Already decided that is not viable to have everyone login to a central machine to run it. I need to automate the copy of module into each machine ,import the module an then run the script.
